mlir/lib/Quantum/Transforms/BufferizableOpInterfaceImpl.cpp

```diff
@@ -0,0 +1,60 @@
+#include "mlir/Dialect/Bufferization/IR/BufferizableOpInterface.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+#include "Quantum/IR/QuantumOps.h"
+#include "Quantum/Transforms/BufferizableOpInterfaceImpl.h"
+
+using namespace mlir;
+using namespace catalyst::quantum;
+
+namespace {
+
+/// Bufferization of tensor.extract. Replace with memref.load.
+struct StateOpInterface
+    : public mlir::bufferization::BufferizableOpInterface::ExternalModel<StateOpInterface,
+                                                    catalyst::quantum::StateOp> {
+  bool bufferizesToMemoryRead(mlir::Operation *op, mlir::OpOperand &opOperand,
+                              const mlir::bufferization::AnalysisState &state) const {
+    return false;
```

Comment:
Should this be true?

Reply:
Yes. It should be true. Thank you.

Reply:
I don't think this is true. What buffer is it reading from?

Reply:
Sorry, yes it should be false, but it should be bufferizesToMemoryWrite, because we will be writing to it, no?

Reply:
The thing the op has no inputs before it is bufferized, and I think these function are used to perform the analysis of which buffers can be reused etc, so I think it neither reads nor writes (to any inputs).

(And for this op we force the allocation of a new buffer everytime anyways.)